extensions/amp-story/1.0/amp-story.js

```diff
@@ -452,11 +464,47 @@ export class AmpStory extends AMP.BaseElement {
       );
     }
 
+    this.applyDesktopAspectRatioAttribute_();
+
     if (this.maybeLoadStoryDevTools_()) {
       return;
     }
   }
 
+  /**
+   * Grab the desktop-aspect-ratio attribute, clamp the value
+   * between 1/2 and 3/4 aspect ratios, and apply it to the root element.
+   * @private
+   */
+  applyDesktopAspectRatioAttribute_() {
+    if (!this.element.hasAttribute('desktop-aspect-ratio')) {
+      return;
+    }
+
+    const desktopAspectRatio =
+      'clamp(' +
+      MIN_CUSTOM_DESKTOP_ONE_PANEL_ASPECT_RATIO +
+      ', ' +
+      this.element.getAttribute('desktop-aspect-ratio').replace(':', '/') +
+      ', ' +
+      MAX_CUSTOM_DESKTOP_ONE_PANEL_ASPECT_RATIO +
+      ')';
+    setImportantStyles(
+      document.querySelector(
+        ':root:not([data-story-supports-landscape]):not([i-amphtml-story-mobile])'
+      ),
```

Comment:
Does this break if it's null? We should probably not rely on classes inserted at runtime, and instead it to the element only when it doesn't support landscape (`if (this.element.hasAttribute("supports-landscape")) {return;}`). The mobile check doesn't work because the size of the screen might change. Also it's not necessary.

Reply:
Done.